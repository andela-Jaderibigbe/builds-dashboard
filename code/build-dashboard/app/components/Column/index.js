import React from 'react';
import PropTypes from 'prop-types';
import styles from './Column.scss';

const Column = ({ className, children }) => {
  className = `${styles.column} ${className}`;
  return <div className={className}>{children}</div>;
};

Column.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default Column;
