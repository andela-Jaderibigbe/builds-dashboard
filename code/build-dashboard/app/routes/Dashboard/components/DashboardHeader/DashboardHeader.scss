@import 'theme/variables';

.changeListCol {
  width: 19%;
  text-align: left;
}

.ownerCol {
  text-align: center;
  width: 9%;
}

.timeCol {
  text-align: center;
  width: 14%;
}

.stateCol {
  text-align: center;
  width: 14%;
}

.statusCol {
  text-align: center;
  width: 11%;
}
