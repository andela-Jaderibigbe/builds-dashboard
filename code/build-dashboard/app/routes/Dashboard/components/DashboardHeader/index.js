import React from 'react';
import TableHeader from 'components/TableHeader';
import TableRow from 'components/TableRow';
import Column from 'components/Column';
import styles from './DashboardHeader.scss';

const DashboardHeader = () => (
  <TableHeader>
    <TableRow>
      <Column className={styles.changeListCol}>
        <span>Changelist / Build</span>
      </Column>
      <Column className={styles.ownerCol}>
        <span>Owner</span>
      </Column>
      <Column className={styles.timeCol}>
        <span>Time Started</span>
      </Column>
      <Column className={styles.stateCol}>
        <span>State</span>
      </Column>
      <Column className={styles.statusCol}>
        <span>Metrics</span>
      </Column>
      <Column className={styles.statusCol}>
        <span>Build</span>
      </Column>
      <Column className={styles.statusCol}>
        <span>Unit Test</span>
      </Column>
      <Column className={styles.statusCol}>
        <span>Functional Test</span>
      </Column>
    </TableRow>
  </TableHeader>
);

export default DashboardHeader;
