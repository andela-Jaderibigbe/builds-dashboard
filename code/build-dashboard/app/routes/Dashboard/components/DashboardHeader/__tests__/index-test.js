import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { expect as chaiExpect } from 'chai';

import Column from 'components/Column';
import DashboardHeader from '../';

describe('<DashboardHeader />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<DashboardHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have correct number of columns', () => {
    const wrapper = shallow(<DashboardHeader />);
    chaiExpect(wrapper).to.exist;
    chaiExpect(wrapper.find(Column)).to.have.length(8);
  });
});
