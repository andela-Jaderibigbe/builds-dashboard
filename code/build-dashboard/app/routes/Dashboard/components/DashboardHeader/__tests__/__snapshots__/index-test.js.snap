// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardHeader /> should render correctly 1`] = `
<div
  className="tableHeader undefined"
>
  <div
    className="tableRow undefined"
  >
    <div
      className="column changeListCol"
    >
      <span>
        Changelist / Build
      </span>
    </div>
    <div
      className="column ownerCol"
    >
      <span>
        Owner
      </span>
    </div>
    <div
      className="column timeCol"
    >
      <span>
        Time Started
      </span>
    </div>
    <div
      className="column stateCol"
    >
      <span>
        State
      </span>
    </div>
    <div
      className="column statusCol"
    >
      <span>
        Metrics
      </span>
    </div>
    <div
      className="column statusCol"
    >
      <span>
        Build
      </span>
    </div>
    <div
      className="column statusCol"
    >
      <span>
        Unit Test
      </span>
    </div>
    <div
      className="column statusCol"
    >
      <span>
        Functional Test
      </span>
    </div>
  </div>
</div>
`;
